seqr/views/utils/dataset_utils.py

```diff
@@ -320,43 +323,29 @@ def _parse_tsv_row(row):
 
 REVERSE_TISSUE_TYPE = {v: k for k, v in TISSUE_TYPE_MAP.items()}
 
-def _parse_outlier_row(row, **kwargs):
+def _parse_outlier_row(row):
     yield row['sampleID'], {mapped_key: row[key] for key, mapped_key in RNA_OUTLIER_COLUMNS.items()}
 
-def _parse_tpm_row(row, sample_id_to_tissue_type=None):
+def _parse_tpm_row(row):
     sample_id = row[SAMPLE_ID_COL]
     if row[TPM_COL] != '0.0' and not sample_id.startswith('GTEX'):
-        prev_tissue = sample_id_to_tissue_type.get(sample_id)
         tissue = row[TISSUE_COL]
         if not tissue:
             raise ValueError(f'Sample {sample_id} has no tissue type')
-        if prev_tissue and prev_tissue != tissue:
-            raise ValueError(f'Mismatched tissue types for sample {sample_id}: {prev_tissue}, {tissue}')
-        sample_id_to_tissue_type[sample_id] = tissue
 
-        parsed = {GENE_ID_COL: row[GENE_ID_COL], 'tpm': row[TPM_COL]}
+        parsed = {GENE_ID_COL: row[GENE_ID_COL], 'tpm': row[TPM_COL], TISSUE_COL: tissue}
         if INDIV_ID_COL in row:
             parsed[INDIV_ID_COL] = row[INDIV_ID_COL]
 
         yield sample_id, parsed
 
-def _check_invalid_tissues(samples, sample_id_to_tissue_type, warnings):
+def _check_invalid_tissues(samples, sample_id_to_tissue_type):
     invalid_tissues = {}
     for sample in samples:
-        tissue_type = TISSUE_TYPE_MAP[sample_id_to_tissue_type[sample.sample_id]]
-        if not sample.tissue_type:
-            sample.tissue_type = tissue_type
-            sample.save()
-        elif sample.tissue_type != tissue_type:
+        tissue_type = sample_id_to_tissue_type[sample.sample_id]
+        if sample.tissue_type != tissue_type:
             invalid_tissues[sample] = tissue_type
 
-    if invalid_tissues:
-        mismatch = ', '.join([
-            f'{sample.sample_id} ({REVERSE_TISSUE_TYPE[expected_tissue]} to {REVERSE_TISSUE_TYPE[sample.tissue_type]})'
-            for sample, expected_tissue in invalid_tissues.items()])
-        message = f'Skipped data loading for the following {len(invalid_tissues)} samples due to mismatched tissue type: {mismatch}'
-        warnings.append(message)
```

Comment:
The `invalid` tissue samples will be filtered out. So no warning messages are required.

Reply:
but we need to continue to warn the user that these are filtered out, that behavior can't change

Reply:
It doesn't filter out any data. It filters out sample models that don't match the tissues in the input file. New sample models will be created for all the data with mismatched tissue types.

Reply:
We no longer have the warning message because we will create a new sample model and load the data in the case of a mismatched tissue type.

Reply:
I thought we still are only supporting one tissue type per sample per file. What if there are multiple tissue types for the same sample in the file?

Reply:
We don't allow multiple tissue types for the same sample in a project. It makes sense because a sample usually is not possible from various tissues.

Reply:
This issue is still open for discussion. Can a sample ID have multiple tissue types?